### ISSUE TITLE
Deploy spinnaker via halyard 1.34.0 failed

### ISSUE THREAD

**Issue by ealebed** [spinnaker/spinnaker#5685]
### Issue Summary:
Hi, I got error, trying deploy spinnaker with halyard 1.34.0:

```
...
Validation in Global:
! ERROR Unable to render template:
apiVersion: apps/v1 # As of Kubernetes 1.16, apps/v1beta2 has been deprecated
kind: Deployment
metadata:
name: spin-{{ name }}
namespace: {{ namespace }}
annotations:
moniker.spinnaker.io/application: '"spin"'
moniker.spinnaker.io/cluster: '"{{ name }}"'
labels:
app: spin
cluster: spin-{{ name }}
app.kubernetes.io/name: {{ name }}
app.kubernetes.io/managed-by: halyard
app.kubernetes.io/part-of: spinnaker
app.kubernetes.io/version: {{ version }}
spec:
replicas: {{ replicas }}
selector:
matchLabels:
app: spin
cluster: spin-{{ name }}
{% if deploymentStrategy != null %}
strategy:
{% if deploymentStrategy.type == "rollingUpdate" and
  deploymentStrategy.strategySettings|length > 0 %}
{{ deploymentStrategy.type }}:
{% for key, value in deploymentStrategy.strategySettings.items() %}
{{ key }} : {{ value }}
{% endfor %}
{% endif %}
type: {% if deploymentStrategy.type == "recreate" %} Recreate {% else %}
  RollingUpdate {% endif %}
{% endif %}
template:
metadata:
annotations:
{% for key, value in podAnnotations.items() %}
"{{ key }}": "{{ value }}"
{% endfor %}
labels:
app: spin
cluster: spin-{{ name }}
app.kubernetes.io/name: {{ name }}
app.kubernetes.io/managed-by: halyard
app.kubernetes.io/part-of: spinnaker
app.kubernetes.io/version: {{ version }}
{% for key, value in podLabels.items() %}
"{{ key }}": "{{ value }}"
{% endfor %}
spec: {{ podSpec }}
TemplateError{severity=FATAL, reason=SYNTAX_ERROR, item=OTHER,
  message='TemplateSyntaxException: java.lang.IncompatibleClassChangeError:
  Inconsistent constant pool data in classfile for class
  com/hubspot/jinjava/lib/filter/Filter. Method 'java.lang.String
  lambda$filter$3(java.lang.Object)' at index 53 is CONSTANT_MethodRef and should
  be CONSTANT_InterfaceMethodRef', fieldName='null', lineno=24, category=UNKNOWN,
  categoryErrors=null}
com.hubspot.jinjava.interpret.TemplateSyntaxException:
  java.lang.IncompatibleClassChangeError: Inconsistent constant pool data in
  classfile for class com/hubspot/jinjava/lib/filter/Filter. Method
  'java.lang.String lambda$filter$3(java.lang.Object)' at index 53 is
  CONSTANT_MethodRef and should be CONSTANT_InterfaceMethodRef
        at
  com.hubspot.jinjava.el.ExpressionResolver.resolveExpression(ExpressionResolver.java:80)
        at
  com.hubspot.jinjava.interpret.JinjavaInterpreter.resolveELExpression(JinjavaInterpreter.java:337)
        at com.hubspot.jinjava.lib.tag.IfTag.evaluateIfElseTagNode(IfTag.java:100)
        at com.hubspot.jinjava.lib.tag.IfTag.interpret(IfTag.java:63)
        at com.hubspot.jinjava.lib.tag.Tag.interpretOutput(Tag.java:29)
        at com.hubspot.jinjava.tree.TagNode.render(TagNode.java:50)
        at com.hubspot.jinjava.lib.tag.IfTag.interpret(IfTag.java:74)
        at com.hubspot.jinjava.lib.tag.Tag.interpretOutput(Tag.java:29)
        at com.hubspot.jinjava.tree.TagNode.render(TagNode.java:50)
        at
  com.hubspot.jinjava.interpret.JinjavaInterpreter.render(JinjavaInterpreter.java:193)
        at
  com.hubspot.jinjava.interpret.JinjavaInterpreter.render(JinjavaInterpreter.java:165)
        at com.hubspot.jinjava.Jinjava.renderForResult(Jinjava.java:198)
        at com.hubspot.jinjava.Jinjava.renderForResult(Jinjava.java:171)
        at com.hubspot.jinjava.Jinjava.render(Jinjava.java:147)
        at
  com.netflix.spinnaker.halyard.core.resource.v1.JinjaTemplatedResource.toString(JinjaTemplatedResource.java:33)
        at
  com.netflix.spinnaker.halyard.deploy.spinnaker.v1.service.distributed.kubernetes.v2.KubernetesV2Service.getResourceYaml(KubernetesV2Service.java:202)
        at
  com.netflix.spinnaker.halyard.deploy.deployment.v1.KubectlDeployer.lambda$deploy$0(KubectlDeployer.java:102)
        at
  com.netflix.spinnaker.halyard.core.DaemonResponse$StaticRequestBuilder.build(DaemonResponse.java:123)
        at
  com.netflix.spinnaker.halyard.core.tasks.v1.TaskRepository.lambda$submitTask$1(TaskRepository.java:48)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: jinjava.javax.el.ELException: java.lang.IncompatibleClassChangeError:
  Inconsistent constant pool data in classfile for class
  com/hubspot/jinjava/lib/filter/Filter. Method 'java.lang.String
  lambda$filter$3(java.lang.Object)' at index 53 is CONSTANT_MethodRef and should
  be CONSTANT_InterfaceMethodRef
        at jinjava.javax.el.BeanELResolver.invoke(BeanELResolver.java:502)
        at
  com.hubspot.jinjava.el.ext.JinjavaBeanELResolver.invoke(JinjavaBeanELResolver.java:65)
        at jinjava.javax.el.CompositeELResolver.invoke(CompositeELResolver.java:397)
        at jinjava.javax.el.CompositeELResolver.invoke(CompositeELResolver.java:397)
        at jinjava.de.odysseus.el.util.SimpleResolver.invoke(SimpleResolver.java:135)
        at
  com.hubspot.jinjava.el.JinjavaInterpreterResolver.invoke(JinjavaInterpreterResolver.java:96)
        at jinjava.de.odysseus.el.tree.impl.ast.AstMethod.eval(AstMethod.java:91)
        at jinjava.de.odysseus.el.tree.impl.ast.AstMethod.eval(AstMethod.java:100)
        at
  jinjava.de.odysseus.el.tree.impl.ast.AstBinary$SimpleOperator.eval(AstBinary.java:31)
        at jinjava.de.odysseus.el.tree.impl.ast.AstBinary.eval(AstBinary.java:110)
        at jinjava.de.odysseus.el.tree.impl.ast.AstBinary$2.eval(AstBinary.java:43)
        at jinjava.de.odysseus.el.tree.impl.ast.AstBinary.eval(AstBinary.java:110)
        at jinjava.de.odysseus.el.tree.impl.ast.AstEval.eval(AstEval.java:51)
        at jinjava.de.odysseus.el.tree.impl.ast.AstNode.getValue(AstNode.java:31)
        at
  jinjava.de.odysseus.el.TreeValueExpression.getValue(TreeValueExpression.java:122)
        at
  com.hubspot.jinjava.el.ExpressionResolver.resolveExpression(ExpressionResolver.java:67)
        ... 19 more
Caused by: java.lang.IncompatibleClassChangeError: Inconsistent constant pool
  data in classfile for class com/hubspot/jinjava/lib/filter/Filter. Method
  'java.lang.String lambda$filter$3(java.lang.Object)' at index 53 is
  CONSTANT_MethodRef and should be CONSTANT_InterfaceMethodRef
        at com.hubspot.jinjava.lib.filter.Filter.filter(Filter.java:53)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native
  Method)
        at
  java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at
  java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at jinjava.javax.el.BeanELResolver.invoke(BeanELResolver.java:500)
        ... 34 more
- Failed to deploy Spinnaker.
```
### Cloud Provider(s):
Kubernetes GKE

### Environment:
Spinnaker 1.18.+ deployed to Kubernetes (V2) GKE cluster via halyard

### Feature Area:
halyard

**Comment by plumpy** [comment]
How are you deploying Halyard? Are you using the pre-built containers?

**Comment by iiro** [comment]
Sorry for commenting a closed issue, but @plumpy - I got the exactly same error (see below) elsewhere (when Spinnaker tries to render a MPTv1 pipeline template):

`TemplateError{severity=FATAL, reason=SYNTAX_ERROR, item=OTHER,
  message='TemplateSyntaxException: java.lang.IncompatibleClassChangeError:
  Inconsistent constant pool data in classfile for class
  com/hubspot/jinjava/lib/filter/Filter. Method 'java.lang.String
  lambda$filter$3(java.lang.Object)' at index 53 is CONSTANT_MethodRef and should
  be CONSTANT_InterfaceMethodRef', fieldName='null', lineno=24, category=UNKNOWN,
  categoryErrors=null}
com.hubspot.jinjava.interpret.TemplateSyntaxException:
  java.lang.IncompatibleClassChangeError: Inconsistent constant pool data in
  classfile for class com/hubspot/jinjava/lib/filter/Filter. Method
  'java.lang.String lambda$filter$3(java.lang.Object)' at index 53 is
  CONSTANT_MethodRef and should be CONSTANT_InterfaceMethodRef
        at
  com.hubspot.jinjava.el.ExpressionResolver.resolveExpression(ExpressionResolver.java:80)
        at`

I was just thinking, can it be that we have a fleet of similar problems elsewhere than what was fixed on this issue...?

I can open a new issue but just wanted to ask you first.
